viz/tap/api/handlers.go

```diff
@@ -112,7 +127,9 @@ func (h *handler) handleTap(w http.ResponseWriter, req *http.Request, p httprout
 	if len(path) == 8 {
 		resource = path[5]
 	} else if len(path) == 10 {
-		resource = path[7]
+		resource = path[8]
+	} else if len(path) == 12 {
+		resource = path[8]
```

Comment:
IIUC, given `type` is now explicitly "labeled" as a query param, it might be cleaner to define `type := p.ByName("type")`. And then if `type` is empty have `resource = "namespaces"`, otherwise `resource = type`. Instead of using the `path` array.

Reply:
Yep I think it'd be good to use the query param for this.

I don't think I'll change it to check if `type` is empty and assume `resource = "namespaces"`. I think it'd be good to keep the length check, and if it's `8` then `resources = "namespaces"`. And it's not `10` or `12`, then throw the error.

Reply:
@alpeb I'm actually going to keep this as is. In order to use `p.ByName("type")`, the URL needs to have the `:type` parameter in the route (`typeRoute := fmt.Sprintf("/apis/%s/watch/namespaces/:namespace/type/%s", gvk.GroupVersion().String(), res.name)`).

Right now we know there routes only for the resource types we give, but if I change this to be `.../type/:type`, then we'd have to check in `handleTap` that it's a valid type. That seems like an unnecessary check to add to avoid just getting the index of the resource type.